### ISSUE TITLE
Crash when profile does not exist anymore

### ISSUE THREAD

**Issue by doc75** [phase1geo/Minder#285]
**Version used**: git master on 2021-01-31

**Scenario to reproduce the issue**:
* Create a new profile
* Create a new minder file
* Select the custom profile
* Save the file
* Close minder
* remove `.local/share/minder` directory (I am under linux)
* Launch minder (+ open file using custom profile if you closed it before exiting minder) => crash (cf. stacktrace below)

**Expected result**:
* Minder should open the file even if profile is not existing anymore

**Workaround**:
* Open the minder file and remove the `theme` tag

<details>
<summary>Stacktrace</summary>

```
Thread 1 "com.github.phas" received signal SIGSEGV, Segmentation fault.
0x00005555556fa639 in theme_make_icon (self=0x555556f15f50) at ../src/Theme.vala:286
286        set_context_color( ctx, get_color( "background" ) );
(gdb) where
#0  0x00005555556fa639 in theme_make_icon (self=0x555556f15f50) at ../src/Theme.vala:286
#1  0x00005555556ffa17 in themes_icons (self=0x555555942710, icons=0x7fffffffb4f0)
    at ../src/Themes.vala:74
#2  0x000055555568a8a2 in map_inspector_update_themes (self=0x555555deb6b0)
    at ../src/MapInspector.vala:264
#3  0x0000555555688049 in _map_inspector_update_themes_themes_themes_changed
    (_sender=0x555555942710, self=0x555555deb6b0) at ../src/MapInspector.vala:51
#4  0x00007ffff7f5c802 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#5  0x00007ffff7f70814 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#6  0x00007ffff7f7bb9e in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#7  0x00007ffff7f7c0d3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#8  0x00005555556ffe10 in themes_add_theme (self=0x555555942710, theme=0x555556f15f50)
    at ../src/Themes.vala:101
#9  0x000055555560c356 in draw_area_load_theme (self=0x555555e65950, n=0x555555eb8ae0)
    at ../src/DrawArea.vala:465
#10 0x000055555560cb26 in draw_area_load (self=0x555555e65950, n=0x5555568f5bb0)
    at ../src/DrawArea.vala:507
#11 0x00005555556085fc in document_load (self=0x555556f071c0) at ../src/Document.vala:101
#12 0x000055555567dfcf in main_window_open_file
    (self=0x555555a6b9e0, fname=0x5555569622a0 "/home/virlet/Data/Perso/musique/methodes/helene_de_montgeroult/Hélène-de-Mongeroult.minder") at ../src/MainWindow.vala:962
#13 0x000055555567dbbb in main_window_do_open_file (self=0x555555a6b9e0) at ../src/MainWindow.vala:946
#14 0x0000555555673563 in _main_window_do_open_file_gtk_button_clicked
    (_sender=0x555555a43c00, self=0x555555a6b9e0) at ../src/MainWindow.vala:185
#15 0x00007ffff7f5ca56 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#16 0x00007ffff7f7bb28 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#17 0x00007ffff7f7c0d3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#18 0x00007ffff75272ae in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#19 0x00007ffff7527318 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#20 0x00007ffff7f5c802 in g_closure_invoke () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#21 0x00007ffff7f70962 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#22 0x00007ffff7f7bb9e in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#23 0x00007ffff7f7c0d3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#24 0x00007ffff7525754 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#25 0x00007ffff77d6ae1 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#26 0x00007ffff7f5ca56 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#27 0x00007ffff7f7bb28 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#28 0x00007ffff7f7c0d3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#29 0x00007ffff75edfcc in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#30 0x00007ffff7f5fc56 in g_cclosure_marshal_VOID__BOXEDv ()
    at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#31 0x00007ffff7f5ca56 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#32 0x00007ffff7f7bb28 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#33 0x00007ffff7f7c0d3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#34 0x00007ffff75eb012 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#35 0x00007ffff75ec65b in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#36 0x00007ffff75ef646 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#37 0x00007ffff75b6bb0 in gtk_event_controller_handle_event () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#38 0x00007ffff777916d in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#39 0x00007ffff77d05ef in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#40 0x00007ffff7f5ca56 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#41 0x00007ffff7f7add1 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#42 0x00007ffff7f7c0d3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#43 0x00007ffff777ac23 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#44 0x00007ffff7636128 in  () at /lib/x86_64-linux-gnu/libgtk-3.so.0
--Type <RET> for more, q to quit, c to continue without paging--
#45 0x00007ffff76383db in gtk_main_do_event () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#46 0x00007ffff731ef79 in  () at /lib/x86_64-linux-gnu/libgdk-3.so.0
#47 0x00007ffff7352106 in  () at /lib/x86_64-linux-gnu/libgdk-3.so.0
#48 0x00007ffff7e70f9d in g_main_context_dispatch () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#49 0x00007ffff7e71220 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#50 0x00007ffff7e71513 in g_main_loop_run () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#51 0x00007ffff763737d in gtk_main () at /lib/x86_64-linux-gnu/libgtk-3.so.0
#52 0x00005555555e36ee in minder_real_activate (base=0x55555585e120) at ../src/Application.vala:97
#53 0x00007ffff7f5ca56 in  () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#54 0x00007ffff7f7bb28 in g_signal_emit_valist () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#55 0x00007ffff7f7c0d3 in g_signal_emit () at /lib/x86_64-linux-gnu/libgobject-2.0.so.0
#56 0x00007ffff715ad88 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#57 0x00007ffff715af42 in g_application_run () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#58 0x00005555555e40b4 in minder_main (args=0x7fffffffdf08, args_length1=1)
    at ../src/Application.vala:154
#59 0x00005555555e40fb in main (argc=1, argv=0x7fffffffdf08) at ../src/Application.vala:138
(gdb)
```
</details>

**Comment by phase1geo** [comment]
Minder is supposed to use the default theme if the custom theme does not exist. If the application is crashing, that is definitely a bug.

**Comment by phase1geo** [comment]
I see that the bug report talks about a "profile".  What is a "profile"?

**Comment by phase1geo** [comment]
I was thinking that a profile was just a named used to represent a theme.  However, a custom theme is saved to the .minder file that is using it.  Deleting the .local/share/minder directory won't have the effect of deleting the theme itself.  If "profile" is the same thing as a theme, I was unable to replicate the bug following the directions above.

**Comment by MartineMin** [comment]
I confirm that Minder crashes when we change the color of a node creating a new theme

**Comment by phase1geo** [comment]
I believe that this issue should now be fixed in the master branch which will be generally available in the 1.12.2 release when it is made.  There was another issue with a different but related issue that I have fixed which should also take care of this issue.  Give it a try and feel free to update this issue.
